doc/Process_Console.md

```diff
@@ -0,0 +1,589 @@
+ Process Console
+ ===============
+ 
+ `process_console` is a capsule that implements a small shell over the UART that allows
+ a terminal to inspect the kernel and control userspace processes.
+
+Commands
+ --------
+
+ This module provides a simple text-based console to inspect and control
+ which processes are running. The console has eleven commands:
+  - `help` - prints the available commands and arguments
+  - `status` - prints the current system status
+  - `list` - lists the current processes with their IDs and running state
+  - `stop n` - stops the process with name n
+  - `start n` - starts the stopped process with name n
+  - `fault n` - forces the process with name n into a fault state
+  - `terminate n` - terminates the running process with name n, moving it to the Terminated state
+  - `boot n` - tries to restart a Terminated process with name n
+  - `panic` - causes the kernel to run the panic handler
+  - `kernel` - prints the kernel memory map
+  - `process n` - prints the memory map of process with name n
```

Comment:
Move this below as a sort of table of contents for the more in depth descriptions and examples of each command.